pkg/daemon/config.go

```diff
@@ -146,6 +153,16 @@ func (output *ptp4lConf) populatePtp4lConf(config *string) error {
 			output.sections = append(output.sections, currentSection)
 		}
 	}
+	if !hasSlaveInterfaceDefined {
+		// No Slave Interfaces defined
+		output.clock_type = event.GM
+	} else if len(output.sections) > 1 {
```

Comment:
This is not true if there is a [global] section.
Suggest move this logic after Line#168, so that a [global] section is always present, and then change this line to
`} else if len(output.sections) > 2 {`